app/javascript/components/search/results/DataRepoStudy.js

```diff
@@ -0,0 +1,66 @@
+/* show a search result entry from Data Repo */
+import React from "react";
+export const descriptionCharacterLimit = 750
+export const summaryWordLimit = 150
+
+import {getDisplayNameForFacet} from "providers/SearchFacetProvider";
+
+export function shortenDescription(textDescription) {
+  const suffixTag = <span className="detail"> ...(continued)</span>
+  const displayedStudyDescription = { __html: textDescription.slice(0, descriptionCharacterLimit) }
+  if (textDescription.length>descriptionCharacterLimit) {
+    return <>
+      <span className="studyDescription" dangerouslySetInnerHTML={displayedStudyDescription}></span>{suffixTag}
+    </>
+  } else {
+    return <><span className = 'studyDescription' dangerouslySetInnerHTML={displayedStudyDescription}></span></>
+  }
+}
+
+/* generate a badge for each matched facet, containing the filter names */
+function facetMatchBadges(study) {
+  const matches = study.facet_matches
+  if (!matches) {
+    return <></>
+  }
+  const matchedKeys = Object.keys(matches)
+    .filter(key => key != 'facet_search_weight')
+  return (<>
+    { matchedKeys.map((key, index) => {
+      const helpText = `Metadata match for ${key}`
+      return (
+        <span key={index}
+              className="badge badge-secondary facet-match"
+              data-toggle="tooltip"
+              title={helpText}>
+          {
+            matches[key].map(filter => {
+              if ('min' in filter) { // numeric facet
+                return `${getDisplayNameForFacet(key)} ${filter.min}-${filter.max} ${filter.unit}`
+              } else {
+                return filter.name
+              }
+            }).join(',')
+          }
+        </span>)
+    })}
+  </>)
+}
+
+export default function DataRepoStudy({ study }) {
```

Comment:
It seems to me there is enough overlap in the `SCPStudy` and `DataRepoStudy` that they could be within one component to reduce some of the duplication, I was making that change in my branch (which is branched off this one) so not blocking but just wanted to mention it.

Reply:
I would say that any React refactoring you want to do in your branch should supersede anything here.  These components were simply a _very_ rough way of getting the TDR result into the UI.

Reply:
Concur.  My two cents is that I would lean towards keeping these as separate components, since they are likely to be distinct visual entities. And since they share a lot of common attributes, the DataRepoStudy component might import a lot of logic and display from the SCPStudy component.  But that's something the implementor can figure out :)